src/dash/models/DashManifestModel.js

```diff
@@ -773,6 +787,56 @@ function DashManifestModel() {
         return utcTimingEntries;
     }
 
+    function getBaseURLsFromElement(node) {
+        let baseUrls = [];
+        let entries = node.BaseURL_asArray || [node.baseUri] || [];
+
+        entries.some(entry => {
+            if (entry) {
+                const baseUrl = new BaseURL();
+                const text = entry.__text || entry;
+
+                baseUrl.url = text;
+
+                // serviceLocation is optional, but we need it in order
+                // to blacklist correctly. if it's not available, use
+                // anything unique since there's no relationship to any
+                // other BaseURL and, in theory, the url should be
+                // unique so use this instead.
+                if (entry.hasOwnProperty('serviceLocation') &&
+                        entry.serviceLocation.length) {
+                    baseUrl.serviceLocation = entry.serviceLocation;
+                } else {
+                    baseUrl.serviceLocation = text;
+                }
+
+                if (entry.hasOwnProperty('dvb:priority')) {
+                    baseUrl.dvb_priority = entry['dvb:priority'];
+                }
+
+                if (entry.hasOwnProperty('dvb:weight')) {
+                    baseUrl.dvb_weight = entry['dvb:weight'];
+                }
+
+                /* NOTE: byteRange, availabilityTimeOffset,
+                 * availabilityTimeComplete currently unused
+                 */
+
+                baseUrls.push(baseUrl);
+
+                if (urlUtils.isRelative(text)) {
+                    // it doesn't really make sense to have relative and
+                    // absolute URLs at the same level, or multiple
+                    // relative URLs at the same level, so assume we are
+                    // done from this level of the MPD
```

Comment:
I suppose it might make sense, but I think/hope that it's probably a contrived example.

This comment, and the accompanying implementation, was based on informative guidelines in DVB A168 11.9.5 which, among other useful information, implies that mixing relative and absolute BaseURL elements at the same level would reduce clarity.